lib/ops/opshandler/opshandler.go

```diff
@@ -2307,62 +2335,25 @@ func (s *WebHandler) needsAuth(fn ServiceHandle) httprouter.Handle {
 
 // GetHandlerContext authenticates the user that made the request and returns
 // the appropriate handler context
-func GetHandlerContext(w http.ResponseWriter, r *http.Request, backend storage.Backend, operator ops.Operator, auth httplib.Authenticator, usersService users.Identity) (*HandlerContext, error) {
-	authCreds, err := httplib.ParseAuthHeaders(r)
+func GetHandlerContext(w http.ResponseWriter, r *http.Request, backend storage.Backend, operator ops.Operator, authenticator users.Authenticator, usersService users.Identity) (*HandlerContext, error) {
+	authResult, err := authenticator.Authenticate(w, r)
 	if err != nil {
-		return nil, trace.Wrap(err, "credentials error")
+		log.WithError(err).Debug("Authentication error.")
+		return nil, trace.AccessDenied("bad username or password") // Hide the actual error.
```

Comment:
When suppressing an error, I think logging of that error should not be debug, IE something high enough that it would actually be present in the logs. Possibly even with a correlation ID, where the correlation ID can be used to find the full error.